superset/views/core.py

```diff
@@ -2398,7 +2398,13 @@ def results(self, key):
                 status=410,
             )
 
-        query = db.session.query(Query).filter_by(results_key=key).one()
+        query = db.session.query(Query).filter_by(results_key=key).one_or_none()
+        if query is None:
+            return json_error_response(
+                "Data could not be retrieved. You may want to re-run the query.",
+                status=410,
```

Comment:
Is this the proper status code? if the query isn't found, then I think 404 might be better

Reply:
I was debating whether to use 404, which I think fits better, or 410 like a previous similar error message used. I'll use 404 to make it easier to differentiate.